### PR TITLE
handle custom docker for none xsoar supported packs

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
[XSUP-25287](https://jira-hq.paloaltonetworks.local/browse/XSUP-25287)

## Description
**validate** will no longer throw errors for custom dockers in case of none xsoar supported pack.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 * Removed the `timezone_offset` argument in the *modeling-rules test* command.
 * Fixed an issue where **lint** failed when importing functions from CommonServerUserPython.
 * The **format** command now will sync hidden parameters with master branch.
+* **validate** will no longer throw errors for custom dockers in case of none xsoar supported pack.
```
Comment (JudahSchwartz) [inline CHANGELOG.md:8]:
```suggestion
* **validate** will only fail on docker related errors if the pack is supported by xsoar.
```

File: demisto_sdk/commands/common/hook_validations/docker.py [hunk 2]
```diff
@@ -58,6 +58,9 @@ def __init__(
         self.code_type = self.get_code_type()
         self.yml_docker_image = self.get_docker_image_from_yml()
         self.from_version = self.yml_file.get("fromversion", "0")
+        self.is_pack_xsoar_supported = (
+            get_pack_metadata(yml_file_path).get("support", "xsoar").lower() == "xsoar"
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/common/hook_validations/docker.py:62]: should the default be xsoar? I imagine we would be more careful with this than customers will be. IMO should change default to cimmunity

File: demisto_sdk/commands/common/hook_validations/docker.py [hunk 3]
```diff
@@ -96,7 +99,12 @@ def is_docker_image_valid(self):
             error_message, error_code = Errors.non_existing_docker(
                 self.yml_docker_image
             )
-            if self.handle_error(error_message, error_code, file_path=self.file_path):
+            if self.handle_error(
+                error_message,
+                error_code,
+                file_path=self.file_path,
+                warning=not self.is_pack_xsoar_supported,
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/common/hook_validations/docker.py:106]: why add to warning? Whats this do?

**Review by JudahSchwartz — COMMENTED** [review]
nice fix!

File: demisto_sdk/commands/common/hook_validations/docker.py (hunk 3 above)
Comment (YuvHayun) [reply demisto_sdk/commands/common/hook_validations/docker.py:106]: It'll print this error as a warning.
I think that if we decide not to fail the validation for none-xsoar supported then we should at least warn about it.

File: demisto_sdk/commands/common/hook_validations/docker.py (hunk 2 above)
Comment (YuvHayun) [reply demisto_sdk/commands/common/hook_validations/docker.py:62]: If no info is found than validate as it was xsoar supported - sounds the safest.

Comment (JudahSchwartz) [reply demisto_sdk/commands/common/hook_validations/docker.py:62]: are we able to assume that customers will have this field set on their own cicd pipelines? @dorschw maybe you know if we can rely on that?

**Review by JudahSchwartz — APPROVED** [review]
Approved pending the discussion about the default pack level. Great work

File: demisto_sdk/commands/common/hook_validations/docker.py (hunk 2 above)
Comment (YuvHayun) [reply demisto_sdk/commands/common/hook_validations/docker.py:62]: Yes, there's a validate to ensure this field exists in the pack_metadata

**Pull request merged**